Dockerfile

```diff
@@ -15,10 +14,17 @@
 
 COPY --from=builder /app .
 
-RUN pip install --upgrade pip && \
-    pip install -r requirements.txt
+# Update and install dependencies with cleanup to reduce image size
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    libmariadb-dev \
+    && rm -rf /var/lib/apt/lists/* \
+    && pip install --upgrade pip \
+    && pip install mysqlclient \
+    && pip install -r requirements.txt
 
 EXPOSE 8080
 
-# Run database migrations and start the Django application
-ENTRYPOINT ["sh", "-c", "python manage.py migrate && python manage.py runserver 0.0.0.0:8080"]
+COPY wait-for-it.sh /app/wait-for-it.sh
+RUN chmod +x /app/wait-for-it.sh
+
+ENTRYPOINT ["sh", "-c", "./wait-for-it.sh mysql:3306 -- python manage.py migrate && python manage.py runserver 0.0.0.0:8080"]
```

Comment:
Be cautious with overriding the `ENTRYPOINT` in your Docker setup, as it might lead to unexpected behavior. Ensure that the `wait-for-it.sh` script is correctly handling the dependencies before starting the application.